### PR TITLE
docs(nx): add more detail to Nx CLI guide

### PR DESCRIPTION
**Pull request by brandonroberts**
_[Please make sure you have read the submission guidelines before posting an PR](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#submit-pr)_

> _Please make sure that your commit message follows our format._

> Example: `fix(nx): must begin with lowercase`

### CODE REVIEW

File: docs/react/guides/cli.md [hunk 1]
```diff
@@ -66,31 +66,270 @@ nx build myapp --prod
...
+
+Generate a dependency graph for the affected projects
+
+```sh
+nx affected:dep-graph
+```
+
+---
+
+### affected
```
Comment (vsavkin) [inline docs/react/guides/cli.md]: Is it actually printing the list of projects?

File: docs/react/guides/cli.md [hunk 2]
```diff
@@ -66,31 +66,270 @@ nx build myapp --prod
...
+```sh
+nx affected:lint --parallel
+```
+
+After running the affected:lint command, to only rerun failed projects:
+
+```sh
+nx affected:lint --only-failed
+```
+
```
Comment (vsavkin) [inline docs/react/guides/cli.md]: Even though you mentioned it, maybe it's worth explicitly showing how to use nx affected --target=lint. Just so folks know that they can run anything with affected

**Review by vsavkin — CHANGES_REQUESTED** [review]
I like it. It's much better than before.

There are a few more things I'd like to mention. 
- Configurations. And that any builder invocation involves project/target/configuration.
- defaultCollection. Why `nx g app myapp` can be the same as `nx g @nrwl/react:app myapp`.
- nx g app --help shows the options

And I'd like to mention that it's extensible:
- That folks can add their own targets (maybe use the run-commands builder as an example)

File: docs/react/guides/cli.md (hunk 1 above)
Comment (vsavkin) [reply docs/react/guides/cli.md]: Does it work for you?

I'm getting "Invalid options provided"

File: docs/react/guides/cli.md [hunk 3]
```diff
@@ -66,31 +66,333 @@ nx build myapp --prod
...
+### affected
+
+Display the help usage for available options to determine the affected projects.
+
+```sh
+nx affected
+```
+
+You can read more about these commands in the guide about [nly rebuilding and retesting what is affected](react/guides/monorepo-affected).
```
Comment (vsavkin) [inline docs/react/guides/cli.md]: nly => only

**Review by vsavkin — CHANGES_REQUESTED** [review]

**Comment by vsavkin** [comment]
@brandonroberts sorry, I think it needs to be rebased.

**Comment by brandonroberts** [comment]
@vsavkin done

**Pull request merged**